manage_arkime/core/price_report.py

```diff
@@ -0,0 +1,79 @@
+from core.capacity_planning import ClusterPlan
+from core.user_config import UserConfig
+from typing import Dict
+from dataclasses import dataclass
+import math
+
+AWS_HOURS_PER_MONTH=730
+AWS_SECS_PER_MONTH=60*60*AWS_HOURS_PER_MONTH
+
+us_east_1_prices: Dict[str, float] = {
+    "t3.small.search": 0.0360 * AWS_HOURS_PER_MONTH,
+    "m5.large.search": 0.1420 * AWS_HOURS_PER_MONTH,
+    "m6g.large.search": 0.1280  * AWS_HOURS_PER_MONTH,
+    "c6g.2xlarge.search": 0.4520 * AWS_HOURS_PER_MONTH,
+    "r6g.large.search": 0.1670 * AWS_HOURS_PER_MONTH,
+    "r6g.2xlarge.search": 0.6690 * AWS_HOURS_PER_MONTH,
+    "r6g.4xlarge.search": 1.3390 * AWS_HOURS_PER_MONTH,
+    "r6g.12xlarge.search": 4.0160 * AWS_HOURS_PER_MONTH,
+    "m5.xlarge": 0.1920 * AWS_HOURS_PER_MONTH,
+
+    "s3-STANDARD-50-GB": 0.023, # https://aws.amazon.com/s3/pricing/
+    "s3-STANDARD-450-GB": 0.022,
+    "s3-STANDARD-REST-GB": 0.021,
+    "ebs-GB": 0.10, # https://aws.amazon.com/ebs/pricing/
+    "gwlb-GB": 0.004, # https://aws.amazon.com/elasticloadbalancing/pricing/?nc=sn&loc=3
+    "gwlbe-GB": 0.0035, # https://aws.amazon.com/privatelink/pricing/
+
+    "fargate": 0.04048 * AWS_HOURS_PER_MONTH, # https://aws.amazon.com/fargate/pricing/
+
+    "trafficmirror": 0.015 * AWS_HOURS_PER_MONTH, # https://aws.amazon.com/vpc/pricing/
+}
+
+
+@dataclass
+class PriceReport:
+    plan: ClusterPlan
+    config: UserConfig
+    prices = us_east_1_prices # ALW - Not sure how to do type hint, was getting errors
```

Comment:
I think the type hint would be `prices: Dict[str, float]`.

Also, TL;DR - would prefer not to set this default value here and instead pass it as an argument into the constructor or set it inside the constructor body.

Factoid - when you set a container with mutable contents (like the dictionary `us_east_1_prices`) as a default value for method (in this case, `__init__()`), it's only evaluated once when the method/class definition is read for the first time.  All instances of the class will share a reference to the same object created during that initial reading.  This can lead to situations where one instance makes changes to the default value that should be local to it but actually affects all instances of the class.  For this reason, folks usually don't set complex types as defaults like this.

In this case, this is supposed to be a constant value that should never change, but the convention would be to either pass it in as an argument to the object's instantiation or explicitly declare it in the `__init__()` method, making the copy local to the instance.

Reply:
You would think, but then you get the error 

```
Traceback (most recent call last):
  File "/Users/andywick/cloud-demo/./manage_arkime.py", line 8, in <module>
    from commands.cluster_create import cmd_cluster_create
  File "/Users/andywick/cloud-demo/manage_arkime/commands/cluster_create.py", line 18, in <module>
    from core.price_report import PriceReport
  File "/Users/andywick/cloud-demo/manage_arkime/core/price_report.py", line 37, in <module>
    class PriceReport:
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/dataclasses.py", line 1021, in dataclass
    return wrap(cls)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/dataclasses.py", line 1013, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/dataclasses.py", line 863, in _process_class
    cls_fields = [_get_field(cls, name, type)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/dataclasses.py", line 863, in <listcomp>
    cls_fields = [_get_field(cls, name, type)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/dataclasses.py", line 747, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'dict'> for field prices is not allowed: use default_factory
```

The reason i did it here was to have a default if non is provided which is what I saw suggested other places. shrug.